darwin/dataset/remote_dataset.py

```diff
@@ -1,83 +1,135 @@
+        image_count : int
+            Dataset size (number of images)
+        progress : int
+            TODO this is not clear, as it should be a float for how it is handled and its never really used
+        client : Client
+            Client to use for interaction with the server
+        """
         self.name = name
         self.slug = slug or name
         self.dataset_id = dataset_id
         self.project_id = project_id
         self.image_count = image_count
         self.progress = progress
-        self._client = client
-
-    def upload_files(self, files: List[str], fps: int = 1):
-        """ A generator where each file is emitted upon upload """
-        yield from self._add_files_to_dataset(files, fps=fps)
-
-    def _add_files_to_dataset(self, filenames: List, fps: int):
-        """Helper function: upload images to an existing remote dataset. """
-        if not filenames:
-            return
-        for filenames_chunk in RemoteDataset.chunk(filenames, 100):
-            images, videos = _split_on_file_type(filenames_chunk)
-            data = self._client.put(
-                endpoint=f"/datasets/{self.dataset_id}",
-                payload={"image_filenames": images,
-                         "videos": [{"fps": fps, "original_filename": video} for video in videos]}
-            )
-
-        for image_file in data["image_data"]:
-            metadata = upload_file_to_s3(self._client, image_file)
-            self._client.put(f"/dataset_images/{metadata['id']}/confirm_upload", payload={})
-            yield
+        self.client = client
+
+    def push(
+        self,
+        blocking: Optional[bool] = True,
+        multi_threaded: Optional[bool] = True,
+        extensions_to_exclude: Optional[List[str]] = None,
+        fps: Optional[int] = mp.cpu_count(),
+        files_to_upload: Optional[List[Path]] = None,
+    ):
+        """Uploads a local project (images ONLY) in the projects directory.
 
-        for video_file in data["video_data"]:
-            metadata = upload_file_to_s3(self._client, video_file)
-            self._client.put(f"/dataset_videos/{metadata['id']}/confirm_upload", payload={})
-            yield
+        Parameters
+        ----------
+        blocking : bool
+            If False, the dataset is not downloaded and a generator function is returned instead
+        multi_threaded : bool
+            Uses multiprocessing to download the dataset in parallel.
+            If blocking is False this has no effect.
+        extensions_to_exclude : list[str]
+            List of extensions to exclude
+        fps : int
+            Number of file per seconds to upload
+        files_to_upload : list[Path]
+            List of files to upload
 
-    @staticmethod
-    def _f(x):
-        """Support function for pool.map() in pull()"""
-        x()
+        Returns
+        -------
+        generator : function
+            Generator for doing the actual uploads. This is None if blocking is True
```

Comment:
Yes it allows for examples and notes (some examples here https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) 

An example of usage is provided in the blocking scenario. 
If someone wants to get the generator instead of the images it should be knowing what is doing anyway (and can see the code provided to have an idea how to pull it)